src/lua/uri.lua

```diff
@@ -280,26 +281,12 @@ local function escape(buf, opts)
     options.unreserved = options.unreserved or RFC3986.unreserved
     options.plus = options.plus or RFC3986.plus
 
-    local pattern = unreserved(options.unreserved)
-    local unreserved = {}
-    for i = 0, 255 do
-        local ch = string.char(i)
-        unreserved[ch] = ch:match(pattern)
-    end
-
-    return (buf:gsub("(.)", function(ch)
-        if options.plus and ch == " " then
-            return "+"
-        elseif unreserved[ch] then
-            return ch
-        else
-            return char_to_hex(ch)
-        end
-    end))
-end
+    local c_str = ffi.new("char[?]", #buf + 1)
+    ffi.copy(c_str, buf, #buf)
+    local c_escaped = builtin.uri_escape(c_str, #buf, options.unreserved,
+                                         options.plus)
```

Comment:
The malloc'ed string is not freed. The same for the `builtin.uri_unescape()` call.